### PR TITLE
Resolution smearing function in refl1d

### PR DESCRIPTION
**Pull request by andped10**
Introduced resolution function (`resolution_function` ) for refl1d  rather than constant (`dQ`).  The code now supports either a constant value or a linear interpolation between known resolution values (should be read from file in a coming PR).

Coming PRs:
- be able pass resolution data loaded from file #130
- adjust refnx to resolution function #128

### CODE REVIEW

File: EasyReflectometry/calculators/refl1d/wrapper.py [hunk 1]
```diff
@@ -157,20 +156,21 @@ def calculate(self, x_array: np.ndarray, model_name: str) -> np.ndarray:
...
             intensity=self.storage['model'][model_name]['scale'],
             background=self.storage['model'][model_name]['bkg'],
         )
         q.calc_Qo = np.linspace(
-            x_array[argmin] - 3.5 * dq_vector[argmin],
-            x_array[argmax] + 3.5 * dq_vector[argmax],
-            21 * len(x_array),
+            q_array[argmin] - 3.5 * dq_vector_normalized_to_refnx[argmin],
+            q_array[argmax] + 3.5 * dq_vector_normalized_to_refnx[argmax],
+            21 * len(q_array),
```
Comment (rozyczko) [inline EasyReflectometry/calculators/refl1d/wrapper.py:173]: Those hard-coded values `3.5` and `21` are very mysterious. Can we get some explanation or maybe make them parameters, defined somewhere at the top of the file?

File: EasyReflectometry/calculators/wrapper_base.py [hunk 2]
```diff
@@ -198,3 +202,21 @@ def get_item_value(self, name: str, key: str) -> float:
...
+        :param resolution_function: The resolution function
+        """
+        self._resolution_function = resolution_function
+
+
+def constant_resolution_function(constant: float) -> Callable[[np.array], float]:
+    return linear_spline_resolution_function([sys.float_info.min, sys.float_info.max], [constant, constant])
+
+
+def linear_spline_resolution_function(q_data_points: np.array, resolution_points: np.array) -> Callable[[np.array], float]:
```
Comment (rozyczko) [inline EasyReflectometry/calculators/wrapper_base.py:218]: please add a description of what this method is doing - it isn't immediately obvious

**Review by henrikjacobsenfys — APPROVED** [review]
Unrelated to the changes in this pull request, there's a potential issue with calc_Qo - see
https://github.com/EasyScience/EasyReflectometryLib/issues/131

File: EasyReflectometry/experiment/model.py [hunk 3]
```diff
@@ -185,11 +191,17 @@ def uid(self) -> int:
     @property
     def _dict_repr(self) -> dict[str, dict[str, str]]:
         """A simplified dict representation."""
+        resolution_values = self._resolution_function([0.1, 0.2])
```
[the remark this answers is not in the document]
Comment (andped10) [reply EasyReflectometry/experiment/model.py]: Changed way it is determined if the resolution function is a constant

File: EasyReflectometry/calculators/wrapper_base.py (hunk 2 above)
Comment (andped10) [reply EasyReflectometry/calculators/wrapper_base.py]: done

File: EasyReflectometry/calculators/refl1d/wrapper.py (hunk 1 above)
Comment (andped10) [reply EasyReflectometry/calculators/refl1d/wrapper.py]: done

**Review by rozyczko — APPROVED** [review]

**Pull request merged**